Client/Frontend/TabBar/TabBarViewController.swift

```diff
@@ -235,6 +235,19 @@ class TabBarViewController: UIViewController, UITextFieldDelegate, UrlViewContro
         }
     }
 
+    override func viewDidLayoutSubviews() {
+        // Due to an apparent bug on the iPhone 6 Plus (bug 1124310), calling textField.selectAll
+        // too early can cause a crash (which we do in response to making it the first responder).
+        // As a workaround, wait until the view layout has happened. Since this method is called
+        // for layout changes, we need to use a boolean flag to make sure we only handle it once.
+        // We could move this to viewDidAppear, but that results in a noticeable delay before the
+        // keyboard appears.
+        if !didInitialLayout {
+            didInitialLayout = true
+            toolbarTextField.becomeFirstResponder()
```

Comment:
Yep, see initial PR comment and the comment in code. <code>viewWillAppear</code> is still too early and crashes. <code>viewDidAppear</code> works, but it's delayed -- there's noticeable lag before the text is highlighted and the keyboard appears.